src/Tasks/Movie/Main.java

```diff
@@ -0,0 +1,196 @@
+
+    public static void main(String[] args) {
+        MovieApp[] movies = new MovieApp[3];
+
+        while (true) {
+            System.out.print("""
+                    Press 1 to input movies
+                    Press 2 to display movies and ratings
+                    Press 3 to input more movies
+                    Press 4 to find statistics
+                    Press 5 to search movie
+                    Press 6 to update rating of the movie
+                    Press 7 to delete a movie
+                    Press 8 to sort movies
+                    Press 0 to exit
+                    """);
+
+            int input = sc.nextInt();
+            sc.nextLine();
+
+            switch (input) {
+                case 1 -> addMovie(movies);
+                case 2 -> listMovie(movies);
+                case 3 -> movies = addAdditional(movies);
+                case 4 -> showStatistics(movies);
+                case 5 -> findMovie(movies);
+                case 6 -> updateRating(movies);
+                case 7 -> movies = deleteMovie(movies);
+                case 8 -> sortMovies(movies);
+                case 0 -> {
+                    return;
+                }
+                default -> System.out.println("Enter valid input");
+            }
+        }
+    }
+
+    public static void addMovie(MovieApp[] movies) {
+        for (int i = 0; i < movies.length; i++) {
+            System.out.println("Enter movie name:");
+            String name = sc.nextLine();
+            System.out.println("Enter rating");
+            double rating = sc.nextDouble();
+            sc.nextLine();
+            movies[i] = new MovieApp(name, rating);
+        }
+    }
+
+    public static MovieApp[] addAdditional(MovieApp[] movies) {
+        int prevlength = movies.length;
+        System.out.print("Number of movies you want to add: ");
+        int number = sc.nextInt();
+        sc.nextLine(); // after int/double there is buffer/empty line
+        MovieApp[] newMovies = new MovieApp[movies.length + number];
+
+        System.arraycopy(movies, 0, newMovies, 0, movies.length);
+
+        for (int i = prevlength; i < newMovies.length; i++) {
+            System.out.println("Enter movie name:");
+            String name = sc.nextLine();
+            System.out.println("Enter rating:");
+            double rating = sc.nextDouble();
+            sc.nextLine();
+            newMovies[i] = new MovieApp(name, rating);
+        }
+
+        return newMovies;
+    }
+
+    public static void listMovie(MovieApp[] movies) {
+        for (MovieApp m : movies) {
+            System.out.println(m);
+        }
+    }
+
+    public static void showStatistics(MovieApp[] movies) {
+        double ratingSum = 0;
+        for (MovieApp movie : movies) {
+            ratingSum += movie.getRating();
+        }
+        System.out.printf("Average rating  is: %.2f%n", ratingSum / movies.length);
+
+        // Min  / Max
+        double max = movies[0].getRating();
```

Comment:
here instead of 4 fields you can use objects itself and then willl be only two fields